tests/unit/s2n_connection_get_negotiated_group_name_test.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "tls/extensions/s2n_extension_list.h"
+#include "tls/extensions/s2n_client_server_name.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_internal.h"
+#include "tls/s2n_tls.h"
+
+#include "crypto/s2n_hash.h"
+
+#include "utils/s2n_socket.h"
+#include "utils/s2n_blob.h"
+
+#if EVP_APIS_SUPPORTED
+    #define S2N_X25519_IF_AVAILABLE "x25519"
+    #define S2N_X25519_KYBER_R3_IF_AVAILABLE "x25519_kyber-512-r3"
+#else
+    #define S2N_X25519_IF_AVAILABLE "secp256r1"
+#define S2N_X25519_KYBER_R3_IF_AVAILABLE "secp256r1_kyber-512-r3"
+#endif
+
+#if S2N_NO_PQ
+    #undef S2N_X25519_KYBER_R3_IF_AVAILABLE
+    #define S2N_X25519_KYBER_R3_IF_AVAILABLE S2N_X25519_IF_AVAILABLE
+    #define S2N_P256_KYBER_R3_IF_AVAILABLE "secp256r1"
+#else
+    #define S2N_P256_KYBER_R3_IF_AVAILABLE "secp256r1_kyber-512-r3"
+#endif
+
+struct negotiated_policy_params {
+    const char *policy_name;
+    const char *expected_group;
+};
+
+// Must be in the same order as security_policy_selection in tls/s2n_security_policies.c
```

Comment:
This seems a bit fragile... Also comments should only be `/* */`, not `//`.

Reply:
1. Fixed comment style. 
2. I was wanting to ensure that new TLS Policy's added in the future (with a potentially new expected group) are also added to this test. Having this test fail when a new TLS Policy is added to `tls/s2n_security_policies.c` but not to this test is a simple way to do that, and having both lists be in the same order makes it easier to compare them.